src/app/shared/services/file-manager/file-manager.service.ts

```diff
@@ -1,34 +1,72 @@
 import { Injectable } from "@angular/core";
 import { Directory, Encoding, Filesystem } from "@capacitor/filesystem";
+import { FileOpener } from "@capacitor-community/file-opener";
 import { Capacitor } from "@capacitor/core";
 import write_blob from "capacitor-blob-writer";
 import { SyncServiceBase } from "../syncService.base";
 import { environment } from "src/environments/environment";
 import { IAssetContents } from "src/app/data";
+import { TemplateActionRegistry } from "../../components/template/services/instance/template-action.registry";
+import { TemplateAssetService } from "../../components/template/services/template-asset.service";
+import { ErrorHandlerService } from "../error-handler/error-handler.service";
 
 @Injectable({
   providedIn: "root",
 })
 export class FileManagerService extends SyncServiceBase {
   cacheName: string;
 
-  constructor() {
+  constructor(
+    private errorHandler: ErrorHandlerService,
+    private templateActionRegistry: TemplateActionRegistry,
+    private templateAssetService: TemplateAssetService
+  ) {
     super("FileManager");
     this.initialise();
   }
 
   private initialise() {
     this.cacheName = environment.deploymentConfig.name;
+    this.registerTemplateActionHandlers();
+  }
+
+  private registerTemplateActionHandlers() {
+    this.templateActionRegistry.register({
+      download: async ({ args }) => {
+        await this.downloadAndOpenTemplateAsset(args[0]);
+      },
+    });
+  }
+
+  private async downloadAndOpenTemplateAsset(relativePath: string) {
+    await this.templateAssetService.ready();
+    const blob = (await this.templateAssetService.fetchAsset(relativePath, "blob")) as Blob;
+    try {
+      // On native devices, download file to native storage and trigger prompt to open with native apps
+      if (Capacitor.isNativePlatform()) {
+        const { localFilepath } = await this.saveFile(blob, relativePath);
+        FileOpener.open({ filePath: localFilepath, openWithDefault: false });
+      }
+      // On web, open the file in the browser, ready for download
+      else {
+        const fileUrl = window.URL.createObjectURL(blob);
```

Comment:
nit(non-blocking)
As noted in https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL_static#memory_management, when creating object urls usually you want some means of disposing of these to free up memory (particularly if creating large video/pdf objects)

Would it maybe be a simpler solution to just download in standard web way (allow browser to download from asset directory and user open from their downloads). There's various tools that can help with this, e.g.
https://www.npmjs.com/package/downloadjs
https://github.com/eligrey/FileSaver.js

There's also various manual ways to achieve similar functionality, think this article mentions a couple
https://blog.logrocket.com/programmatically-downloading-files-browser/